lib/dsc-lib/src/functions/parse_cidr.rs

```diff
@@ -0,0 +1,274 @@
+use crate::DscError;
+use ipnetwork::IpNetwork;
+use rust_i18n::t;
+use serde_json::{json, Value};
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct ParseCidr {}
+
+impl Function for ParseCidr {
+    fn get_metadata(&self) -> FunctionMetadata {
+        FunctionMetadata {
+            name: "parseCidr".to_string(),
+            description: t!("functions.parseCidr.description").to_string(),
+            category: vec![FunctionCategory::Object],
+            min_args: 1,
+            max_args: 1,
+            accepted_arg_ordered_types: vec![vec![FunctionArgKind::String]],
+            remaining_arg_accepted_types: None,
+            return_types: vec![FunctionArgKind::Object],
+        }
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("{}", t!("functions.parseCidr.invoked"));
+
+        let cidr_string = args[0].as_str().unwrap();
+
+        // Validate that the input contains a CIDR prefix (contains '/')
+        if !cidr_string.contains('/') {
+            return Err(DscError::FunctionArg(
+                "parseCidr".to_string(),
+                t!("functions.parseCidr.invalidCidr", cidr = cidr_string).to_string(),
+            ));
+        }
+
+        let network = cidr_string.parse::<IpNetwork>().map_err(|_| {
+            DscError::FunctionArg(
+                "parseCidr".to_string(),
+                t!("functions.parseCidr.invalidCidr", cidr = cidr_string).to_string(),
+            )
+        })?;
+
+        let result = match network {
+            IpNetwork::V4(net) => {
+                let network_addr = net.network();
+                let broadcast_addr = net.broadcast();
+                let first_usable = if net.prefix() == 32 {
+                    network_addr
+                } else {
+                    let first_ip = u32::from(network_addr);
+                    std::net::Ipv4Addr::from(first_ip + 1)
+                };
+                let last_usable = if net.prefix() == 32 {
+                    broadcast_addr
+                } else {
+                    let last_ip = u32::from(broadcast_addr);
+                    std::net::Ipv4Addr::from(last_ip - 1)
+                };
+
+                json!({
+                    "network": network_addr.to_string(),
+                    "netmask": net.mask().to_string(),
+                    "broadcast": broadcast_addr.to_string(),
+                    "firstUsable": first_usable.to_string(),
+                    "lastUsable": last_usable.to_string(),
+                    "cidr": net.prefix()
+                })
+            }
+            IpNetwork::V6(net) => {
+                let network_addr = net.network();
+                let broadcast_addr = net.broadcast();
+
+                json!({
+                    "network": network_addr.to_string(),
+                    "netmask": net.mask().to_string(),
+                    "broadcast": broadcast_addr.to_string(),
```

Comment:
Based on this example: https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-cidr#examples, ipv6 doesn't return `broadcast` property